test/cloudtest/test_storaged_shared_fate.py

```diff
@@ -0,0 +1,219 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import subprocess
+import time
+from textwrap import dedent
+
+from materialize.cloudtest.application import MaterializeApplication
+
+CLUSTER_SIZE = 8
+NUM_SOURCES = 4
```

Comment:
In addition to sources, the other concept that exercises storage clusters are sinks. So sinks need to be created and validated as part of this test to ensure complete coverage (that is, the sink will keep emitting after a disruption and no duplicate events will be emitted). 

There are two ways to validate sinks in testdrive:

1. For lower volumes of data, you can try using the `$ kafka-verify-data` testdrive command, as seen for example at the bottom of the `test/testdrive/uuid.td` file. What this command does is connect to Kafka and fetch the records that the sink has placed in the sink topic. Those records are upserts, hence the presence of `before` and `after` sections in the json lines after `$ kafka-verify-data`

2. A bit more convoluted, but will work with larger volumes of data -- ingest the data emitted from the sink back into Mz so that you can validate it with a simple `SELECT` statement. For an example, see line 81 and above in `materialized-views.td`

Larger volumes of data is what we need here -- the sink should produce multiples of `CLUSTER_SIZE` records in order to make sure that each process in the storage cluster is participating (right now maybe only 1 of the processes will actually emit to the sink, but this limitation will probably be lifted eventually)